test/webhook-manager.js

```diff
@@ -0,0 +1,146 @@
+const request = require('supertest')
+const ngrok = require('ngrok')
+
+const { getAccessTokenHeaders } = require('./auth')
+
+const noop = () => Promise.resolve()
+
+// Utility to test external service webhooks integration into Stelace
+// Two modes of functioning:
+// - test webhooks for real via Ngrok
+// - simulate webhooks by fetching and creating events
+//    (suitable if webhooks cannot be created via API and one cannot have access to Ngrok paid plan
+//     for reserved subdomains)
+class WebhookManager {
+  /**
+   * @param {Object}   params
+   * @param {Object}   params.t - AVA test object
+   * @param {Boolean}  [params.isWebhookSimulated = true]
+   *
+   * @param {Function} [params.simulateWebhookSince(lastFetchTimestamp)] - required if isWebhookSimulated is true
+   *   poll events from external service and hit Stelace webhook endpoint
+   *   should return the last event timestamp for next fetch
+   *   this timestamp will be injected at the next call of the function `simulateWebhookSince`
+   * @example
+   *   const processedEventIds = {}
+   *
+   *   async simulateWebhookSince(lastFetchTimestamp) {
+   *     // we recommend to apply deduplication on events
+   *     // because events created at the same second can be not available at the same time
+   *     // via Events API
+   *     const events = await exampleSdk.events.list({ gt: lastFetchTimestamp })
+   *
+   *     for (const event of events) {
+   *       if (processedEventIds[event.id]) continue
+   *
+   *       await request(t.context.serverUrl)
+   *         .post(webhookUrl)
+   *         .send(event)
+   *         .set({ authorization: `Basic ${basicAuth}` })
+   *         .expect(200)
+   *
+   *       processedEventIds[event.id] = true
+   *     }
+   *
+   *     const timestamps = events.map(e => e.created)
+   *     const maxTimestamp = _.max(timestamps) - 1 // to be sure to fetch processed events
```

Comment:
`any unprocessed events`?

And `1` is like a magic number here. Is it supposed to be milliseconds, seconds? Maybe you need some variable name for this safety margin.

I know it’s just an example but this is also a reference for implementation so this has to be easy to grasp.